internal/host/logscmd.go

```diff
@@ -0,0 +1,46 @@
+package host
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/models"
+)
+
+type logsCmd struct {
+	baseCmd
+	instructionConfig InstructionConfig
+}
+
+func NewLogsCmd(log logrus.FieldLogger, instructionConfig InstructionConfig) *logsCmd {
+	return &logsCmd{
+		baseCmd:           baseCmd{log: log},
+		instructionConfig: instructionConfig,
+	}
+}
+
+func (i *logsCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	// added to run upload logs if install command fails
+	if host.LogsCollectedAt != strfmt.DateTime(time.Time{}) {
+		return nil, nil
+	}
```

Comment:
It is possible that hosts logs were already sent to service but installer didn't finish yet?
In this case you may have host logs already but installer failed, in this case you won't get the installer-gather logs ?!